src/base/static/components/molecules/form-field-types/number-field.js

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "react-emotion";
+
+import { NumberInput } from "../../atoms/input";
+
+const NumberField = styled(NumberInput)(props => ({
+  border: props.theme.input.border,
+  padding: props.theme.input.padding,
+  width: "100%",
+  boxSizing: "border-box",
+  backgroundColor: props.hasAutofill
+    ? props.theme.input.autofillBgColor
+    : props.theme.input.defaultBgColor,
+}));
```

Comment:
You might be having trouble with the NumberInput atom above because this NumberField is a styled component. I don't think this should be a styled component. I think you can do something like so:

```javascript
const NumberField = props => {
  return (
    <NumberInput hasAutofill={true} {...props}/>
  )
}
```

Regarding your earlier comment:

>I might be missing something basic, but the emotion docs suggest passing through the classname here:

The issue I have with class names is not a technical one. It's that class names muddy up the API that our atoms are exposing. So I don't think we should allow the client - in this case the NumberField molecule - to arbitrarily alter the styles of the NumberInput atom. I think should adhere to an API that the NumberInput atom exposes via its props. Of course, I could be wrong on this approach, and I'm open to other ideas :)

It might be easier to chat on this tomorrow as well.